test/client_service_base.test.js

```diff
@@ -0,0 +1,290 @@
+      sinon.replace(service, 'sendRequest',
+          sinon.fake.returns(function() {}));
+    }
+
+    describe('registerContract', () => {
+      it('should throw an error when contractBytes is not a Uint8Array',
+          async () => {
+            const service = new ClientServiceBase(
+                ledgerClient, protobuf, clientProperties);
+            try {
+              await service.registerContract('contract1', 'foo', 'wrongType');
+            } catch (e) {
+              assert.instanceOf(e, IllegalArgumentError);
+            }
+          },
+      );
+      it('should work as expected', async () => {
+        const mockedContractRegistrationRequest = {
+          setContractId: function() {},
+          setContractBinaryName: function() {},
+          setContractByteCode: function() {},
+          setContractProperties: function() {},
+          setCertHolderId: function() {},
+          setCertVersion: function() {},
+          setSignature: function() {},
+        };
+        const mockedProtobuf = {
+          ContractRegistrationRequest: function() {
+            return mockedContractRegistrationRequest;
+          },
+        };
+        const service = new ClientServiceBase(
+            ledgerClient, mockedProtobuf, clientProperties);
+        genericEllipticSignatureSigner(service);
+        const mockSpyContractRegistrationRequest = sinon.spy(mockedProtobuf,
+            'ContractRegistrationRequest');
+        const mockSpySetContractBinaryName = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setContractBinaryName');
+        const mockSpySetContractId = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setContractId');
+        const mockSpySetContractByteCode = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setContractByteCode');
+        const mockSpySetContractProperties = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setContractProperties');
+        const mockSpySetCertHolderId = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setCertHolderId');
+        const mockSpySetCertVersion = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setCertVersion');
+        const mockSpySetSignature = sinon.spy(
+            mockedContractRegistrationRequest,
+            'setSignature');
+        const request = await service.registerContract('contractId', 'foo',
+            new Uint8Array([1, 2, 3]), clientProperties);
+        assert(mockSpySetContractId.calledOnce);
+        assert(mockSpyContractRegistrationRequest.calledOnce);
+        assert(mockSpySetContractBinaryName.calledOnce);
+        assert(mockSpySetContractByteCode.calledOnce);
+        assert(mockSpySetContractProperties.calledOnce);
+        assert(mockSpySetCertHolderId.calledOnce);
+        assert(mockSpySetCertVersion.calledOnce);
+        assert(mockSpySetSignature.calledOnce);
```

Comment:
So as to make the test more robust, would it be possible to assert that the spied on method was called with an expected parameter ?
For example, for the `mockSpySetContractId`, can we verify the method was called once with the parameter `'contractId'`.
If it is possible, let's do the same thing for other tests.